interpreter/cling/lib/Interpreter/DeclCollector.h

```diff
@@ -116,6 +118,10 @@ namespace cling {
 
     // dyn_cast/isa support
     static bool classof(const clang::ASTConsumer*) { return true; }
```

Comment:
You likely need: `static bool classof(const clang::ASTDeserializationListener*) { return true; }` and drop the `dynamic_cast` assert.